config/src/network_id.rs

```diff
@@ -54,6 +54,15 @@ impl NetworkContext {
         self.role
     }
 
+    #[cfg(any(test, feature = "testing", feature = "fuzzing"))]
+    pub fn mock_with_peer_id(peer_id: PeerId) -> std::sync::Arc<Self> {
```

Comment:
I know it probably doesn't matter but I suggest something more specific:

```
pub fn validator_for_testing(peer_id: PeerId) -> Self {
  Self {
    network_id: NetworkId::Validator,
    ...
  }
}

```

Reply:
Hmm... see the mock below, of course apparently I wrote that one too... what better way to have a mock that is ... completely bogus.